bioptim/interfaces/acados_interface.py

```diff
@@ -181,13 +202,29 @@ def __set_costs(self, ocp):
                         )
                         self.mayer_costs = vertcat(self.mayer_costs, mayer_func_tp(ocp.nlp[i].X[0]))
                         if J[0]["target"] is not None:
-                            self.y_ref_end.append([J[0]["target"]])
+                            self.y_ref_end.append(J[0]["target"])
                         else:
-                            self.y_ref_end.append([np.zeros((J[0]["val"].numel(), 1))])
+                            self.y_ref_end.append([0] * (J[0]["val"].numel()))
 
                     else:
                         raise RuntimeError("The objective function is not Lagrange nor Mayer.")
 
+                # parameter as mayer cost
+                for j, J in enumerate(ocp.J):
+                    mayer_func_tp = Function(f"cas_mayer_func_{i}_{j}", [ocp.nlp[i].X[-1]], [J[0]["val"]])
```

Comment:
Technically, this would work for any objective stored in J (even not params one). Shall we add a test in order to process only params objective terms ?. I suggest you add a comment to explicit that in the code and we will continue the discussion on pariterre's branch